### DISCUSSION TITLE
fucntion with open()

### DISCUSSION THREAD

**Discussion by ghost** [pybricks/support#1371] [Q&A]
I'm working on a reinforcement learning project and would like to save my q table values in a txt file. 
However the function with open() doesn't work. 
Please help
```python
with open('file.txt', 'r') as f:
    contents = f.read()
    print(contents)`
```
Error Message:
```console
Traceback (most recent call last):
  File "test.py", line 36, in <module>
OSError: 2
```

**Comment by dlech** [comment]
File APIs only work with Pybricks on LEGO MINDSTORMS EV3. All of the Powered Up hubs currently don't have file system support.

For small amounts of data, you can use `hub.system.storage()` on Powered Up hubs.

https://docs.pybricks.com/en/latest/hubs/primehub.html#pybricks.hubs.PrimeHub.system.storage

**Comment by ghost** [comment]
oh thanks, thats a start, my dictionary has over 1'000 states with over 50 actions, little bit too much ;-)
are there plans for open() function in the future?
